src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -202,6 +201,27 @@ public static bool IsNumericType(this SpecialType specialType)
             }
         }
 
+        /// <summary>
+        /// Checks if a type is considered a "built-in integral" by CLR.
+        /// </summary>
+        public static bool IsIntegralType(this SpecialType specialType)
```

Comment:
>IsIntegralType [](start = 27, length = 14)

Please add a compiler test that will fail once this change is merged with the native ints feature. That way we'll be reminded/forced to add logic here.

Reply:
Hum, I got confused. Native ints have been merged already.
Tagging @cston to confirm whether this method should handle them.

---
In reply to: [412454952](https://github.com/dotnet/roslyn/pull/43492#discussion_r412454952) [](ancestors = 412454952)

Reply:
I reviewed this method before merging the native integers feature, and didn't find a caller that might have a native integer type.

---
In reply to: [412468506](https://github.com/dotnet/roslyn/pull/43492#discussion_r412468506) [](ancestors = 412468506,412454952)